src/utils/hoc.js

```diff
@@ -20,27 +20,38 @@ export let withLoader = Component =>
   _.flow(
     wrapDisplayName('withLoader', Component),
     observer
-  )(({ Loader = StripedLoader, node, ...props }) => (
-    <Loader loading={node && node.updating}>
-      <Component node={node} {...props} />
-    </Loader>
-  ))
+  )(({ Loader, ...props }) => {
+    let { theme = {}, node } = props
+    Loader = Loader || theme.Loader || StripedLoader
```

Comment:
You're assuming there's a `theme` being passed down, which is not a good assumption. Just wrap this with `withTheme`. Also don't pass the `theme` down in the props to the Component since it can do so itself.

Reply:
Wouldn't it make sense to assume that if `theme` is a prop on the component that `withTheme` has already been used higher up? I don't see any cases where we have `withLoader` without `withTheme` preceding it.

Reply:
Perhaps right now we don't, but making assumptions about the calling environment increases coupling and causes subtle bugs.

Reply:
By that logic, shouldn't this also be wrapped with `withNode`? What I am getting at is that dependencies are not necessarily a bad thing, and are actually already assumed.

Reply:
You're right I hadn't realized the node dependency. In that case we should add `withNode` to the mix here as well. It's just not good UX imo to have a `withLoader` that in order to use, you need to remember to also wrap it with `withNode` and `withTheme`. If it always needs to be done, then might as well export it with those two HOCs already applied.

Reply:
I understand what you are saying, but I don't agree. I think we can revisit this later, but for now, I think this is the most consistent with the current state of the codebase.